api/controllers/StatsController.js

```diff
@@ -130,6 +192,21 @@ module.exports = {
           }
         }
       });
+
+      games.forEach((game) => {
+        const relevantSeason = seasons.find((season) => {
+          // Find season this match took place during
+          return dayjs(game.updatedAt).isBetween(dayjs(season.startTime), dayjs(season.endTime));
+        });
```

Comment:
I'd like to DRY this up since it's used in for both games and matches above, and ideally refactor it to use a binary search so that these loops each run in O(Nlog(N)) instead of `O(N^2)`

Unfortunately I don't see a way to reduce the repetition of 

```
        if (!relevantSeason) {
          return;
        }
```

which both also use

Reply:
Marking this so I can look later :)

Reply:
Can this be moved to be a part of the match logic above?  Since you already have the `relevantSeason` for a match, you can then associate the match's games with that season as well.  Would also DRY it up since you're only looping through everything once

Reply:
@byarmis the issue there is that we're looping through the matches in one spot and the games in another, so it's not clear to me how to combine them. `relevantSeason` is relative to whichever game or match you're looking at i.e. it's the season in which that game or match took place. So it would be different for the games vs the matches.

Granted we could query for all the games associated with each match and process those with a nested for loop, but the issue there is that we only have matches for ranked games so that would skip all the unranked ones. That's why we have the query for all finished games as separate from the matches and why the loop is processed as a sibling rather than a nested loop: the games include ranked and unranked while the matches are only ranked

Reply:
This speaks to a suggestion you made earlier @byarmis to associate every game (including unranked) with a match. That might clean up this processing, (though it introduces its own complexities), but it would entail a larger rework

Reply:
Ahh that's a good point, I didn't remember that matches only happened for ranked games.

It doesn't appear to be the most common paradigm used in this code base, but you could also maybe have a seasons class that, as an instance variable, has a list of seasons and a `.findSeason()` method that takes in a date and returns a season so the sorting and binary searching only need to happen in one place.

Reply:
Yeah I'd definitely like to sort the seasons and replace this `.find()` with a binary search. Since getting the seasons requires a query we want to make sure we only query for the seasons once and that this search method just processes them in memory.

My plan was to define it as a function that takes the sorted array of seasons and the date as arguments and to define it as a sails helper since that's generally how we've organized the reusable backend utilities. A static method on a singleton class is an interesting idea though.

Reply:
I've pulled the season lookup into  `getSeasonFromTimestamp()` above to DRY this up and utilize binary search